tests/test/projection.rs

```diff
@@ -232,6 +232,49 @@ fn normalize_gat2() {
     }
 }
 
+#[test]
+fn normalize_gat_const() {
```

Comment:
This is a good start, but there's a few more tests that I would like to see before this lands. We can discuss on Zulip, but generally I'm interested that testing *specific* values works.

Reply:
Yeah, I was a little bit lazy with the tests, I just wanted to confirm I took the right path with the implementation?